x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -148,6 +148,15 @@
                                 RoleDescriptor.IndicesPrivileges.builder()
                                     .indices("apm-*")
                                     .privileges("read", "read_cross_cluster").build(),
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices("logs-apm*")
+                                    .privileges("read", "read_cross_cluster").build(),
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices("metrics-apm*")
+                                    .privileges("read", "read_cross_cluster").build(),
+                                RoleDescriptor.IndicesPrivileges.builder()
+                                    .indices("traces-apm*")
+                                    .privileges("read", "read_cross_cluster").build(),                            
```

Comment:
To me, the index names are named in a way that can easily clash with user-defined ones.
I wonder whether it would be possible to nest under the existing `apm*` pattern?

Also, I wonder what's the purpose of the `apm_user` role in this case? Shouldn't it be updated as well.

From a Security perspective, you're granting read access to the whole of Kibana for the whole of APM data.
There's no separation by APM deployments or by users in Kibana!

This is not really an ES concern. I will politely pass it to @legrego .

LGTM provided you get confirmation from Kibana Security.

Reply:
@sqren can you elaborate on why the Kibana system user needs access to these new indices? The PR states that this is for end-users, but I was under the assumption that end-users still needed to have their own index access granted ([docs](https://www.elastic.co/guide/en/kibana/current/apm-app-reader.html)).

Reply:
The kibana system user needs access to the APM indices so we can collect telemetry for this. It already has access tok `apm-*`
I've updated the PR description to include this.

Reply:
> end-users still needed to have their own index access granted

Under fleet users will not be able to configure where data is written to. It will always go to `logs-apm*, metrics-apm*, traces-apm*`

Reply:
> To me, the index names are named in a way that can easily clash with user-defined ones.

Agreed. I know it was mentioned in https://github.com/elastic/kibana/issues/87501, but I also find it strange that there isn't a `-` or `.`, or any other delimiting character in the new index patterns.

I understand that clashing with user-defined indices is currently a problem with the `apm-*` pattern we grant access to today, but I also thought (perhaps incorrectly) this was supposed to be a short-term solution, so I worry about continuing this pattern without more context.

@kobelb I'm sorry to drag you into this, but I know you helped drive this the first time around, and shared concerns about using this as a model going forward. On the surface, this isn't any worse than what we're already doing for `apm-*`, but I wanted to make sure you weren't aware of any alternatives that we could explore. I know the now defunct pulse team was going to investigate this, but I'm not sure how far they got, if at all.

> Also, I wonder what's the purpose of the apm_user role in this case? Shouldn't it be updated as well.

I wonder this too.

Reply:
> To me, the index names are named in a way that can easily clash with user-defined ones.

I think this was decided by the fleet team in collaboration with APM Server team. I'm not sure this decision can easily be changed with less than a week to FF.

> Also, I wonder what's the purpose of the apm_user role in this case? Shouldn't it be updated as well.

Probably just me misunderstanding, but I already did update the `apm_user` with access to the fleet indices.

Reply:
I removed the access to the kibana system user to limit the scope of this PR and ensure it gets merged before FF next week.